### DISCUSSION TITLE
Raspberry Pi Zero 2 W won't act as a USB Drive despite successfully completing

### DISCUSSION THREAD

**Discussion by flootenkerp** [marcone/teslausb#838] [Q&A] [answered]
I can't seem to understand why my Raspberry Pi won't behave as a USB once I go through all the steps. I did the following:

I downloaded teslausb-20231012-bullseye and extracted the 2023-10-12-teslausb image file. Using the Raspberry Pi Imager, I flash my Raspberry Pi Zero 2 W with the image using the following options: RASPBERRY PI ZERO 2 W, Use Custom (Open downloaded image file), and select the target microSD. I say no to the customization of OS settings. 

Once it's successfully flashed, I remount the microSD and drop my teslausb_setup_variables.conf file, which was populated with the requisite archive info and Wifi information. I then insert the microSD into the Raspberry Pi, plug the microUSB into the center USB port of the Raspberry Pi, and plug it into my laptop. 

I wait a bit, then go to the Command Prompt and successfully SSH into pi@teslausb.local. I then tail the teslausb-headless-setup.log and receive the all done message. The Raspberry Pi then starts doing the constant double blinks every second. At this point, the laptop nor the car registers it as a USB when remounted, and when I run df -h when ssh'd into the Raspberry Pi, the TeslaCam folder does not appear as one of the mounted folders. I've been at it for hours and I don't understand what I'm doing wrong. I've even used many different cables to rule out a cable issue. Please assist, and thanks in advance for any help.

**Comment by marcone** [comment, marked as answer]
If it's doing the double blink, it should be acting as a USB drive. You can check in the archive loop log to see if it says "Connected usb to host".
From what you describe it sounds like either all the cables you tried were charge-only cables, or you plugged them into the wrong port on the Pi. You can rule out any problems with TeslaUSB by plugging the Pi without the sd card into a Windows PC, and going to Device Manager. Under "Other devices", there should be a "BCM2710 Boot" entry. If you see that, then your cable and the Pi's USB port are OK.

**Reply by flootenkerp** [reply]
Yeah, all 4 microUSB cables I found in my desk drawer all happened to be power only 😅

I bought a new USB-A to microUSB data transfer cable today and, sure enough, it worked. I just tested it and the footage from the car syncs to the share folder on my network. Thanks so much for the response! This is my first time messing with Raspberry Pi so I very much appreciate it.
